odoo/tests/common.py

```diff
@@ -835,25 +835,23 @@ def url_open(self, url, data=None, timeout=10):
         return self.opener.get(url, timeout=timeout)
 
     def _wait_remaining_requests(self):
-        t0 = int(time.time())
-        for thread in threading.enumerate():
-            if thread.name.startswith('odoo.service.http.request.'):
-                join_retry_count = 10
-                while thread.isAlive():
-                    # Need a busyloop here as thread.join() masks signals
-                    # and would prevent the forced shutdown.
-                    thread.join(0.05)
+        join_retry_count = 10
+        while join_retry_count:
+            threads = [t for t in threading.enumerate() if t.name.startswith('odoo.service.http.request')]
+            if not threads:
+                return
+            for t in threads:
+                t.join(0.05)
+                if t.isAlive():
                     join_retry_count -= 1
-                    if join_retry_count < 0:
-                        self._logger.warning("Stop waiting for thread %s handling request for url %s",
-                                        thread.name, getattr(thread, 'url', '<UNKNOWN>'))
-                        break
-                    time.sleep(0.5)
-                    t1 = int(time.time())
-                    if t0 != t1:
-                        self._logger.info('remaining requests')
-                        odoo.tools.misc.dumpstacks()
-                        t0 = t1
+                    self._logger.info('Waiting for thread %s handling request for url %s', t.name, getattr(t, 'url', '<UNKNOWN>'))
+            time.sleep(0.5)
+
+        for t in threads:
+            if t.isAlive():
```

Comment:
This can probably be removed since the enumerate is only going to return living threads and we're updating it every time.

Reply:
Yes but as there is a last chance (during the last sleep) for the remaining threads to suicide, it's possible that t is not alive any more.

Reply:
Good point.